server/issuers.go

```diff
@@ -42,7 +47,29 @@ func (c *Server) getLatestIssuer(issuerType string) (*Issuer, *handlers.AppError
 			Code:    500,
 		}
 	}
-	return &(*issuer)[0], nil
+	issuerByCohorts := breakCohorts(issuer)
+	lastIssuer := &(*issuerByCohorts)[issuerCohort]
+	if len(*lastIssuer) == 0 {
+		return nil, &handlers.AppError{
+			Message: "Cohort for this issuer not found",
+			Code:    404,
+		}
+	}
+	return &(*lastIssuer)[0], nil
+}
+
+// breaks a slice of []Issuer into a 2D slice with one slice per cohort.
+// This creates an empty slice with two slices, mainly [[],[]], one per cohort.
+// Then we iterate through all issuers, and position each issuer in the slice
+// corresponding to their cohort. Note: this only works for two cohorts. Maybe
+// we want to use a constant defining the number of cohorts?
+func breakCohorts(ss *[]Issuer) (ret *[][]Issuer) {
+	a := make([][]Issuer, 2)
+	for _, s := range *ss {
+		i := s.IssuerCohort
+		a[i] = append(a[i], s)
+	}
+	return &a
```

Comment:
Alternatively what we can do here is only request issuers of a given cohort / issuer type from the database.  E.g. we change the query in `fetchIssuers` to something like this
```
select * from issuers where issuer_type = $1 and cohort = $2
```

That way we don't need to manage picking the right cohort in the application code.  I think this would also generalize to any number of cohorts.  What do you think @iquerejeta ?

Reply:
I thought about this. The problem of requiring a cohort when fetching issuers is that at the time of redemption you need to fetch the two lists
```go 
issuer0 = fetchIssuers(issuerType, issuerCohort = 0)
issuer1 = fetchIssuers(issuerType, issuerCohort = 1)
```
Then we need to verify if the token is valid over the issuers in `issuer1`, and if that fails, check if the token is valid over the issuers in `issuer0`. The motivation of doing it as above is that when calling `fetchIssuers`, the resulting array will contain both issuers of cohort 0 and 1, ordered by expiration date. So we will check if the tokens are valid with the most recent issuers, rather than trying first a cohort, and then another. 

I agree, however, that `breakCohorts` is not very elegant. What do you think?
What do you think?

Reply:
- [ ] Decide on `breakCohorts` design

Reply:
I think it would be better to do this on the database level.  You make a good point, so I think it's appropriate to create a new `fetchIssuersByCohort` function that accepts a cohort as a parameter instead.

Reply:
- [x]